yt_dlp/utils.py

```diff
@@ -2048,9 +2048,11 @@ def url_or_none(url):
     return url if re.match(r'^(?:(?:https?|rt(?:m(?:pt?[es]?|fp)|sp[su]?)|mms|ftps?):)?//', url) else None
 
 
-# TODO
 def request_to_url(req):
-    if isinstance(req, compat_urllib_request.Request):
+    # NOTE: import is now here to avoid making diff. can be moved to top of file
```

Comment:
pretty sure we can't cause circular import. would be better to move to networking package